src/zos-char-util.cc

```diff
@@ -538,27 +538,25 @@ int __file_needs_conversion_init(const char *name, int fd) {
       close(fd);
       return 0;
     }
-    if (cnt > 8) {
-      int ccsid;
-      int am;
-      unsigned len = strlen_ae((unsigned char *)buf, &ccsid, cnt, &am);
-      if (ccsid == 1047 && len == cnt) {
-        if (no_tag_read_behaviour == __NO_TAG_READ_DEFAULT_WITHWARNING) {
-          if (name) {
-            len = strlen(name) + 1;
-            char filename[len];
-            _convert_e2a(filename, name, len);
-            dprintf(2, "Warning: File \"%s\" is untagged and seems to contain "
-                       "EBCDIC characters\n", filename);
-          } else {
-            dprintf(2, "Warning: File (null) is untagged and seems to contain "
-                       "EBCDIC characters\n");
-          }
+    int ccsid;
+    int am;
+    unsigned len = strlen_ae((unsigned char *)buf, &ccsid, cnt, &am);
+    if (ccsid == 1047 && len == cnt) {
+      if (no_tag_read_behaviour == __NO_TAG_READ_DEFAULT_WITHWARNING) {
+        if (name) {
+          len = strlen(name) + 1;
```

Comment:
minor thing but i would typically have len be the strlen and then allocate the buffer to len+1 since you don't need to convert a 0 to a 0. 
But perhaps just me.